chain/src/note_source.rs

```diff
@@ -87,3 +87,18 @@ impl std::fmt::Debug for NoteSource {
         }
     }
 }
+
+impl std::fmt::Display for NoteSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            NoteSource::Transaction { id } => {
+                f.write_fmt(format_args!("NoteSource::Transaction({})", hex::encode(id)))
+            }
+            NoteSource::Genesis => f.write_fmt(format_args!("NoteSource::Genesis")),
+            NoteSource::FundingStreamReward { epoch_index } => f.write_fmt(format_args!(
+                "NoteSource::FundingStreamReward({})",
+                epoch_index
+            )),
+        }
+    }
+}
```

Comment:
I think we should probably drop this impl (for the reason below), but if we did keep it, rather than printing out a Rust-style enum, we should just print the hex-formatted bytes.  The point of the `NoteSource` is to be able to stuff some extra data into something that looks and feels like a transaction hash.